bundle/edu.gemini.qv.plugin/src/main/scala/edu/gemini/qv/plugin/QvContext.scala

```diff
@@ -45,11 +46,11 @@ case class QvContext(peer: Peer, dataSource: DataSource, source: ObservationProv
   def timezone = site.timezone()
 
   // ===== Data providers / filters used in this context ====
-  private val _nonSidProvider = new PositionProvider(this, source)
-  private val _mainFilter = FilterProvider(_nonSidProvider)
-  private val _tableFilter = FilterProvider(_mainFilter)
-  private val _selection = SelectionProvider(_tableFilter)
-  
+  private lazy val _nonSidProvider = new PositionProvider(this, source)
+  private lazy val _mainFilter = FilterProvider(this, _nonSidProvider)
+  private lazy val _tableFilter = FilterProvider(this, _mainFilter)
+  private lazy val _selection = SelectionProvider(this, _tableFilter)
```

Comment:
I've shied away from `lazy` values because of the complicated semantics and because usually it isn't clear that you're gaining that much, if anything.  Are these things expensive and usually not needed, or do you think we should just embrace `lazy`?